docs/db-migration-guide.ipynb

```diff
@@ -0,0 +1,137 @@
+    "mysql --host=$MYSQL_DB_IP --user=$MYSQL_DB_USER --password=$MYSQL_DB_PASSWORD < katib-db.sql\n",
+    "\n",
+    "# relate katib-db-manager and new MySQL database charm\n",
+    "juju relate katib-db-manager:relational-db katib-db-mysql:database"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## KFP DB migration"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "vscode": {
+     "languageId": "shellscript"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# obtain username and password of existing MariadDB database from DB relation\n",
+    "# in Charmed Kubeflow 1.6 and 1.7 username used is `root` and password is specified in `mysql` relation by\n",
+    "# 'root_password'\n",
+    "KFP_DB_MYSQL_RELATION_ID=$(juju show-unit kfp-db/0 | yq '.[] | .relation-info | select(.[].endpoint == \"mysql\") | .[0] | .relation-id')\n",
+    "KFP_DB_USER=root\n",
+    "KFP_DB_PASSWORD=$(bash -c \"juju run --unit kfp-db/0 'relation-get -r $KFP_DB_MYSQL_RELATION_ID - kfp-db/0' | grep root_password\" | awk '{print $2}')\n",
+    "KFP_DB_IP=$(juju show-unit kfp-db/0 | yq '.[] | .address')\n",
+    "\n",
+    "# ensure that there are no new connections are made and that dababase is not altered\n",
+    "# remove relation between kfp-api charm and MariaDB charm\n",
+    "juju remove-relation kfp-api kfp-db\n",
+    "\n",
+    "# connect to unit's IP address using mysql client and verify that all data is intact\n",
+    "mysql --host=$KFP_DB_IP --user=$KFP_DB_USER --password=$KFP_DB_PASSWORD\n",
+    "\n",
+    "# create backup of all databases file using `mysqldump` utility, username, password, and unit's IP address\n",
+    "# onbtained earlier\n",
+    "mysqldump --host=$KFP_DB_IP --user=$KFP_DB_USER --password=$KFP_DB_PASSWORD --column-statistics=0 --databases mlpipeline  > kfp-db.sql\n",
+    "\n",
+    "# deploy new MySQL database charm\n",
+    "juju deploy mysql-k8s kfp-db-mysql --channel 8.0/stable --trust\n",
+    "\n",
+    "# obtain username and password of new MySQL database from DB relation\n",
+    "# note that since relation is not yet established it is impossible to retrieve this information from that relation\n",
+    "# the relation can only be established after data is retored into new database\n",
+    "MYSQL_DB_USER=$(juju run-action kfp-db-mysql/0 get-password --wait | yq '.[] | .results.username')\n",
+    "MYSQL_DB_PASSWORD=$(juju run-action kfp-db-mysql/0 get-password --wait | yq '.[] | .results.password')\n",
+    "MYSQL_DB_IP=$(juju show-unit kfp-db-mysql/0 | yq '.[] | .address')\n",
+    "\n",
+    "# connect to new DB using username, password, and unit's IP address and restore database from backup\n",
+    "mysql --host=$MYSQL_DB_IP --user=$MYSQL_DB_USER --password=$MYSQL_DB_PASSWORD < kfp-db.sql\n",
+    "\n",
+    "# relate kfp-api and new MySQL database charm\n",
+    "juju relate kfp-api:relational-db kfp-db-mysql:database"
+   ]
```

Comment:
Should we remove the *-db application from the model after we are done with the migration? For instance`juju remove-application kfp-db`

Reply:
Good point. Will add.